prow/jobs.md

```diff
@@ -70,7 +70,90 @@ that matches `trigger` will suffice. This is useful if you want to make one
 command that reruns all jobs. If unspecified, the default configuration makes
 `/test <job-name>` trigger the job.
 
-### Pod Utilities
+## Standard Triggering and Execution Behavior for Jobs
+
+When configuring jobs, it is necessary to keep in mind the set of rules Prow has
+for triggering jobs, the GitHub status contexts that those jobs provide, and the
+rules for protecting those contexts on branches.
+
+### Triggering Jobs
+#### Trigger Types
+
+`prow` will consider three different types of jobs that run on pull requests
+(presubmits):
+
+ 1. jobs that run unconditionally and automatically. All jobs that set
+     `always_run: true` fall into this set.
+ 2. jobs that run conditionally, but automatically. All jobs that set
+    `run_if_changed` to some value fall into this set. 
+ 3. jobs that run conditionally, but not automatically. All jobs that set
+    `run_always: false` and do not set `run_if_changed` to any value fall
+    into this set and require a human to trigger them with a command.
+
+By default, jobs fall into the third category and must have their `always_run` or
+`run_if_changed` configured to operate differently.
+
+In the rest of this document, "a job running unconditionally" indicates that the
+job will run even if it is normally conditional and the conditions are not met.
+Similarly, "a job running conditionally" indicates that the job runs if all of its
+conditions are met.
+
+#### Triggering Jobs With Comments
+
+A developer may trigger jobs by posting a comment to a pull request that contains
+one or more of the following phrases:
+ - `/test job-name` : When posting `/test job-name`, any jobs with matching triggers
+   will be triggered unconditionally.
+ - `/retest` : When posting `/retest`, two types of jobs will be triggered:
+   - all jobs that have run and failed will run unconditionally
+   - any not-yet-executed automatically run jobs will run conditionally 
+ - `/test all` : When posting `/test all`, all automatically run jobs will run
+   conditionally.
+   
+Note: is is possible to configure a job's `trigger` to match any of the above keywords
+(`/retest` and/or `/test all`) but this behavior is not suggested as it will confuse
+developers that expect consistent behavior from these commands. More generally, it is
+possible to configure a job's `trigger` to match any command that is otherwise known
+to Prow in some other context, like `/close`. It is similarly not suggested to do this.
+
+#### Posting GitHub Status Contexts
+
+Jobs that run will always post a status context to the commit under test in GitHub.
+Jobs that run conditionally but do not match the content of the pull request will
+_not_ post "Skipped" status contexts to the pull request.
```

Comment:
Any chance we can flag stuff which will be irrelevant to everyone in a year (aka people expecting a skip status to be posted because we did it this way in 2015)? Aka: TODO(fejta): delete the skipped comment in 2020.

Reply:
I tried to word this as something that will continue to be true, not a change from before.

Reply:
Think this still doesn't have a note on when its safe to remove? But not trying to hold up the PR for that.